src/main/java/org/jenkinsci/plugins/deployment/DeploymentTrigger.java

```diff
@@ -67,13 +73,13 @@ public void checkAndFire(DeploymentFacet facet) {
                         if (b!=null) {
                             // pass all the current parameters if we can
                             ParametersAction action = b.getAction(ParametersAction.class);
-                            job.scheduleBuild(job.getQuietPeriod(), new UpstreamDeploymentCause(b), action);
+                            parameterizedJobMixIn.scheduleBuild2(5, action);
```

Comment:
I think you are still loosing some information here, since `new UpstreamDeploymentCause(b)` is not passed to `scheduleBuild2`.